lib/monitor.js

```diff
@@ -0,0 +1,43 @@
+'use babel'
+
+import log from './log'
+import commandLog from './command-log'
+
+function hostIP() {
+  return LEARN_IDE_HOST_IP
+}
+
+function clearHostIP() {
+  LEARN_IDE_HOST_IP = null
+}
+
+export default function monitor(terminal, subscriptions) {
+  subscriptions.add(atom.commands.onDidDispatch(({type}) => {
+    if (type === 'application:quit') { log.add({event: 'quit'}) }
```

Comment:
worked reliably on mac... not so much on windows. so yeah, I need to figure out some way to try and guarantee this.